Reduce Matview storage size and create times

tony-sappe opened the pull request:
- Removes many unnecessary columns from universal_transaction_matview
- Removed unused indexes
- Changed `vacuum analyze` to `analyze`
- Removed the matview clustering

- [x] Run new matview scripts on Dev

dpb-bah commented on usaspending_api/awards/models_matviews.py:
```diff
@@ -7,73 +7,55 @@
     uri = models.TextField()
     piid = models.TextField()
     federal_action_obligation = models.DecimalField(
-        max_digits=20, db_index=True, decimal_places=2, blank=True,
```
Don't see the corresponding `federal_action_obligation` index being removed in the json?

tony-sappe replied:
Django isn't managing this "model" so no need for it listed in code

dpb-bah commented on usaspending_api/database_scripts/matview_generator/universal_transaction_matview.json:
```diff
@@ -103,266 +91,192 @@
       "columns": [{"name": "\"pop_congressional_code\""}]
-    },{
-      "name": "latest",
-      "columns": [{"name": "\"latest_transaction_id\""}]
```
Don't see the corresponding `latest_transaction_id` being removed from a model?

tony-sappe replied:
The column was removed in a previous PR. This just removes the index

dpb-bah commented on usaspending_api/database_scripts/matview_generator/universal_transaction_matview.json:
```diff
@@ -103,266 +91,192 @@
-      "columns": [{"name": "\"awarding_toptier_agency_abbreviation\"", "order": "DESC NULLS LAST"}]
-    },{
-      "name": "awarding_subtier_agency_abbreviation",
-      "where": "awarding_subtier_agency_abbreviation IS NOT NULL",
```
Were these removed from a previous PR? I don't see the corresponding columns being removed in this one.

tony-sappe replied:
The JSON file is a little confusing. all of these dicts are describing indexes. Since there are no queries on the agency abbreviations the indexes won't be used

dpb-bah commented on usaspending_api/database_scripts/matview_generator/universal_award_matview.json:
```diff
@@ -103,7 +105,9 @@
-    "  latest_transaction.\"action_date\" >= '2007-10-01'"
+    "  latest_transaction.\"action_date\" >= '2007-10-01'",
+    "ORDER BY",
+    "  latest_transaction.\"action_date\" DESC"
```
Also why are we adding `order by`s? Shouldn't this just add performance given this is just populating/updating the matviews?

tony-sappe replied:
adding order by to *hopefully* structure the matview for better performance when gathering hundreds/thousands of rows. Since the time_period is a range, it helps to have those columns as close as possible on disk. The clustering was removed which helped performance-wise but took way too much time when creating the matviews

tony-sappe commented:
@dpb-bah is there a good location to store a list of removed columns and indexes?

dpb-bah commented:
It can be provided here in a comment. I just would like a list for future reference if someone wonders why these columns got removed.

tony-sappe commented:
## Columns changes
- universal_award_matview.sql
        - Added `keyword_string`
        - Added `award_id_string`

- universal_transaction_matview.sql
        - Added `keyword_string`
        - Added `award_id_string`
        - Removed `transaction_id`
        - Removed `pop_location_id`
        - Removed `pop_county_name`
        - Removed `face_value_loan_guarantee`
        - Removed `original_loan_subsidy_cost`
        - Removed `awarding_agency_id`
        - Removed `awarding_agency_code`
        - Removed `awarding_agency_name`
        - Removed `funding_agency_id`
        - Removed `funding_agency_code`
        - Removed `funding_agency_name`
        - Removed `recipient_location_id`
        - Removed `recipient_location_state_name`
        - Removed `recipient_location_county_name`


## Indexes changes
- summary_transaction_month_view.sql
        - Removed `psc2` (`psc_code` col removed previous PR)
- universal_award_matview.sql
        - Removed `piid`     (query replaced by `keyword_string`)
        - Removed `gin_pid`  (query replaced by `keyword_string`)
        - Removed `fain`     (query replaced by `keyword_string`)
        - Removed `gin_fain` (query replaced by `keyword_string`)
        - Removed `uri`      (query replaced by `keyword_string`)
        - Removed `gin_uri`  (query replaced by `keyword_string`)
        - Added `keyword_search`
        - Added `award_id_string`

- universal_transaction_matview.sql
        - Removed `transaction_id`
        - Removed `gin`          (query replaced by `keyword_string`)
        - Removed `gin_pid`      (query replaced by `keyword_string`)
        - Removed `ordered_gin`  (query replaced by `keyword_string`)
        - Removed `fain`         (query replaced by `keyword_string`)
        - Removed `gin_fain`     (query replaced by `keyword_string`)
        - Removed `ordered_fain` (query replaced by `keyword_string`)
        - Removed `uri`          (query replaced by `keyword_string`)
        - Removed `ordered_uri`  (query replaced by `keyword_string`)
        - Removed `gin_uri`      (query replaced by `keyword_string`)
        - Removed `latest_transaction_id`
        - Removed `awarding_toptier_agency_abbreviation`
        - Removed `awarding_subtier_agency_abbreviation`
        - Removed `funding_toptier_agency_abbreviation`
        - Removed `funding_subtier_agency_abbreviation`
        - Removed `psc`  (col removed previous PR)
        - Removed `psd`  (col removed previous PR)
        - Removed `product_or_service_description`
        - Removed `gin_product_or_service_description`
        - Removed `naics_description`
        - Removed `face_value_loan_guarantee`
        - Removed `original_loan_subsidy_cost`
        - Added `keyword_search`
        - Added `award_id_string`
